sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +177,163 @@ def _can_handle_query(cls, *query):
         boolean
             answer as to whether client can service the query
         """
-        chkattr = ['Time', 'Instrument', 'SatelliteNumber']
+        chkattr = ["Time", "Instrument", "SatelliteNumber"]
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() in ('xrs', 'goes'):
+            if x.__class__.__name__ == "Instrument" and x.value.lower() in (
+                "xrs",
+                "goes",
+            ):
                 return all(chklist)
         return False
+
+
+class SUVIClient(GenericClient):
+
+    def _get_goes_sat_num(self, date):
+        """
+        Determines the best satellite number for a given date.
+
+        Parameters
+        ----------
+
+        date : `astropy.time.Time`
```

Comment:
We don't want to use the fancy time doc string magic?

Reply:
We should.

Reply:
Like this: https://github.com/sunpy/sunpy/blob/master/sunpy/time/time.py#L258